src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,146 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *  This class represents the list of all tasks.
+ *  It also houses some list operations.
+ *
+ * @author Ryan Tian Jun.
+ */
+public class TaskList {
+    private static List<Task> taskList = new ArrayList<Task>();
+
+    public TaskList() {
+
+    }
+
+    /**
+     * Adds a Task to the List.
+    */
+    public void add(Task task) {
+        taskList.add(task);
+    }
+
+    /**
+     * Searches for Tasks containing a search query.
+     *
+     * @return returns List of all Tasks that contain the Query.
+     */
+    public String search(String query) {
+        List<Task> results = new ArrayList<>();
+        String result = "Here's what I found!: \n";
+
+        for (Task task : taskList) {
+            if (task.getDescription().contains(query)) {
+                results.add(task);
+            }
+        }
+
+        if (results.size() == 0) {
+            return "No matches found!";
+        } else {
+            int counter = 1;
+            for (Task resultFound : results) {
+                String printResult = counter + "." + resultFound + "\n";
+                counter++;
+                result += printResult;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Prints past commands.
+     *
+     * @return returns a printout of all the past user commands.
+     */
+    public static String userCommands() {
```

Comment:
Would it be better if you chose a more intuitive name for this method?

Reply:
Agreed! Will change to listCommand